docs: :memo: using `build` on CLI guide (#100)
seedcase-project/seedcase-flower: docs/guide/cli.qmd

For more details about what you can do with `view`, see `seedcase-flower
view --help`.

## Building documentation from Data Package metadata

In addition to viewing metadata in the terminal, Flower converts
Data Package metadata into a set of plain text files that is more human-friendly than the `datapackage.json`. These files can be

> used to build a documentation site which can make your Data Package's

Reviewer: ```suggestion
used to build a website for the documentation which can make your Data Package's
```

metadata more easily discoverable and readable. Generating this set of

> files can be achieved with the `build` command:

Reviewer: ```suggestion
files with the `build` command:
```

```{.bash filename="Terminal"}
seedcase-flower build
```

This command generates one new file:
`docs/index.qmd`.

> The `qmd` extension indicates that this file contains

Reviewer: ```suggestion
The `qmd` extension means that this file is a
```

[Quarto](https://quarto.org/)-flavored Markdown. While this is the
default text format, you can change it by
specifying your own custom templates as detailed in [the configuration
section of this guide](/docs/design/interface/config.qmd).

> Later in this guide, we will also see how we can generate

Reviewer: ```suggestion
Later in this guide, you will also see how you can generate
```

Later in this guide, you will also see how you can generate
more than one file by using different styles with `build`.

> <!--TODO change link when how-to section exists -->

Reviewer: ```suggestion
```

The PR should be done soon, so we can add it now.

### Changing input and output directory

By default, `build` will look for a `datapackage.json` file in the
current directory, but you can specify a different URI if desired (just
like with `view`):

```{.bash filename="Terminal"}
seedcase-flower build another/directory/datapackage.json
```

To generate files in a different location,
use the `--output-dir` flag with a folder name. For example:

```{.bash filename="Terminal"}
seedcase-flower build --output-dir my-dir
```

This will generate `my-dir/index.qmd` instead of `docs/index.qmd`.

### Styling the output

Just like the `view` command, the `build` command accepts a

> `--style` option, where you can choose to organize the metadata

Reviewer: ```suggestion
`--style` option. This let's you choose how the metadata are organized
```

For example, to split the metadata for each resource in the Data Package
into its own file, you can run the following command:

```{.bash filename="Terminal"}

> seedcase-flower build --style quarto-resource-listing gh:seedcase-project/example-seed-beetle

Author: Good catch, not sure why I changed this here... I think maybe I copied the line from the `view` section and forgot to remove this part.

```

::: callout-important
If you try to use a style that worked fine with `view` for the `build`
command, you might be wondering why you received an error. This is

> because `view` and `build` does not support the same styles, and you

Reviewer: ```suggestion
because `view` and `build` do not support the same styles, and you
```

`view` and `build` support different styles, so a style that works with
`view` may cause an error with `build` and vice versa.

<!-- TODO: Add overview of this or link to an overview -->

:::

> As the default output style `quarto-one-page`, this style also creates the file `docs/index.qmd`. In addition to this, it creates

Reviewer: ```suggestion
Like the default output style is `quarto-one-page`, the `quarto-resource-listing` style creates `docs/index.qmd` but it also creates
```

For more details about what you can do with `view`, see `seedcase-flower
view --help`.

## Building documentation from Data Package metadata

In addition to viewing metadata in the terminal, Flower can organize
Data Package metadata into a set of plain text files. These files can be
used to build a documentation site which can make your Data Package's
metadata more easily discoverable and readable. Generating this set of
files can be achieved with the `build` command:

```{.bash filename="Terminal"}
seedcase-flower build
```

When you run this command, you will see that there is one new file
generated: `docs/index.qmd`. <!--TODO Is this the correct location? -->
The `qmd` extension indicates that this file contains
[Quarto](https://quarto.org/)-flavored Markdown. While this is the

> default text format outputted from Flower, you can change it by

Reviewer 2: ```suggestion
[Quarto](https://quarto.org/)-flavored Markdown. While this is the
default text format, you can change it by
```

specifying your own custom templates as detailed in [the configuration
section of this guide](/docs/design/interface/config.qmd).

<!--TODO change link when how-to section exists -->

### Changing input and output directory

By default, `build` will look for a `datapackage.json` file in the
current directory, but you can specify a different URI if desired (just
like with `view`):

```{.bash filename="Terminal"}
seedcase-flower build another/directory/datapackage.json
```

If you want the generated files to end up in another location, you can
use the `--output-dir` flag followed by a folder name. For example, to
create `my-dir/index.qmd` instead of `docs/index.qmd`, you would run the
following command.

```{.bash filename="Terminal"}
seedcase-flower build --output-dir my-dir
```

### Styling the output

> Just like with the `view` command, the `build` command also accepts the

Reviewer 2: ```suggestion
Just like the `view` command, the `build` command accepts a
```

`--style` option, where you can choose to organize the metadata
according to one of the [built-in
styles](/docs/design/interface/outputs.html#built-in-styles).
<!-- TODO: point link above to guide for styles, or ref docs, not design -->
For example, to split the metadata for each resource in the Data Package
into its own file, you can run the following command:

```{.bash filename="Terminal"}
seedcase-flower build --style quarto-resource-listing
```

::: callout-important
If you try to use a style that worked fine with `view` for the `build`
command, you might be wondering why you received an error. This is
because `view` and `build` do not support the same styles, and you
need to make sure you use the correct set of styles for the respective

> command.

Author: Haha, much more terse, nice!
